pkg/build/api/types.go

```diff
@@ -218,6 +218,11 @@ type BuildStatus struct {
 
 	// Config is an ObjectReference to the BuildConfig this Build is based on.
 	Config *kapi.ObjectReference
+
+	// outputDockerImageDigest is the digest of the built Docker image. The
+	// digest uniquely identifies the image in the registry to which it was
+	// pushed.
+	OutputDockerImageDigest string
```

Comment:
Put this in a struct that parallels spec output - probably `status.output.to.imageDigest` where output is a struct, to is a pointer to a struct, and to is only set when an image was output.